beohms: added recipe

xdizzaster opened the pull request:
Added recipe for BeOhms Ohm's Law calculator.

fbrosson commented:
Thanks @xdizzaster for your PR!
I cannot test it right now, but I think it would be moving the Makefile to the upstream repo.

@pulkomandy, do you think the Makefile @xdizzaster wrote for BeOhm could be added to https://github.com/HaikuArchives/BeOhms/ (with changes, if needed) ?
This would allow to simplify this recipe. What do you think?
Thanls!

xdizzaster commented:
If needed, I could create a PR to add the Makefile to the BeOhms repo.

fbrosson commented on haiku-apps/beohms/beohms-0.1.recipe:
```diff
@@ -0,0 +1,51 @@
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	beohm$secondaryArchSuffix = $portVersion
```
Since your recipes builds fine on all primary architectures, you don't need to have support for secondary architectures.

This means you ~~can~~ should  remove all occurences of $secondaryArchSuffix and {secondaryArchSuffix}.

fbrosson commented on haiku-apps/beohms/beohms-0.1.recipe:
```diff
@@ -0,0 +1,51 @@
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	makefile_engine
```
I think we usually put makefile_engine on the top of the list, before the cmd: lines.

fbrosson commented on haiku-apps/beohms/beohms-0.1.recipe:
```diff
@@ -0,0 +1,51 @@
+BUILD()
+{
+	cp $portDir/additional-files/Makefile $sourceDir/src/Makefile
+	cd src
```
You might wish to simplify this a bit, by changing the cp and cd like this:
```
	cd src
	cp $portDir/additional-files/Makefile .
```

fbrosson commented on haiku-apps/beohms/beohms-0.1.recipe:
```diff
@@ -0,0 +1,51 @@
+{
+	mkdir -p $appsDir
+
+	cp src/bin/BeOhms $appsDir/BeOhms
```
No need to repeat the name of the target, since you are not renaming it.
```
	cp src/bin/BeOhms $appsDir
```
or
```
	cp src/bin/BeOhms $appsDir/
```
if you frefer.

Wait, there is even a much better alternative:
```
	install -d $appsDir
	install -t $appsDir src/bin/BeOhms
```
(It will replace the mkdir and the cp.)

fbrosson commented on haiku-apps/beohms/beohms-0.1.recipe:
```diff
@@ -0,0 +1,51 @@
+DESCRIPTION="BeOhms is a calculator that is based of of Ohm's Law. This \
+calculator solves either voltage, resistance, or current, given two of the \
+variables."
+HOMEPAGE="https://github.com/HaikuArchives/BeOhms/"
```
Here you can drop the trailing slash ;-)

fbrosson commented:
Hmm, I forgot to say that it might be desirable to rename the recipe to beohms-0.0\~git.recipe (and the patchset to beohms-0.0\~git.patchset) or anything you wish, but ending with a "~ git", since there are no release tags yet.)

@korli, @pulkomandy, what would you recommend?

@xdizzaster, for your convenience, here are the commands to use, should you wish to rename the recipe and patchset files.
```
cd haikuports/haiku-apps/beohms

git mv beohms-0.1.recipe beohms-0.0~git.recipe
git commit --amend --no-edit beohms-0.1.recipe beohms-0.0~git.recipe

cd patches

git mv beohms-0.1.patchset beohms-0.0~git.patchset
git commit --amend --no-edit beohms-0.1.patchset beohms-0.0~git.patchset
```
(Please let me know if you are already confortable with these commands, so that I leave them to you ;-)

scottmc commented:
Please send a pull request for BeOhms.  Also decompile the rsrc file down into an rdef and remove the rsrc file, and update the makefile accordingly.  See https://www.haiku-os.org/documents/dev/compile_them_resources

korli commented:
@fbrosson I suggested this page for information about beohms: http://testou.free.fr/www.beatjapan.org/mirror/www.be.com/beware/Utilities/BeOhms.html which states 0.1. Might be incorrect but better than 0.0.

fbrosson commented:
Thanks @korli for the pointers.
@xdizzaster, I've just changed the git commands (on my previous message) to use 0.1~git.
But I don't know if it is still a good idea to rename the recipe, since now we know it is 0.1.

xdizzaster commented:
We are getting version 0.1 from git and we only know it's version 0.1 from an outside source of git. I think version 0.1~git fits our situation the best.

xdizzaster commented:
Changed some stuff @fbrosson suggested. Removed the patchfile and Makefile, as once https://github.com/HaikuArchives/BeOhms/pull/2 is approved, they will not be needed.

waddlesplash commented:
Merged.

xdizzaster commented:
Updated to download the latest commit after @waddlesplash merged HaikuArchives/BeOhms#2

fbrosson commented on haiku-apps/beohms/beohms-0.1~git.recipe:
```diff
@@ -0,0 +1,47 @@
+LICENSE="Public Domain"
+REVISION="1"
+commit="dfdb28d7282aba7d7d616460e85692e68e27a762"
+SOURCE_URI="https://github.com/HaikuArchives/BeOhms/archive/$commit.tar.gz"
```
I don't know if this is a good idea, but we can define SOURCE_URI with:
```
SOURCE_URI="$HOMEPAGE/archive/$commit.tar.gz"
```

xdizzaster replied:
In the other recipes I wrote, I didn't do that, even when I could, so, for the sake of consistency, I think it would be better to leave it as it is.
